### ISSUE TITLE
Instructor: Feedback Result Page: Can see responses that instructors are not allowed to see

### ISSUE THREAD

**Issue by Haozhe321** [TEAMMATES/teammates#8599]
**Environment**

`master` branch commit at `49386fe`

**Steps to reproduce**
1. Login as instructor 
2. Go to a feedback session and add a question
3. Go to custom visibility option and uncheck everything, you should see below
![image](https://user-images.githubusercontent.com/26651556/37038350-829b165e-218f-11e8-8fbf-1b226b7d18d4.png)
4. Save question and login to student account to answer the question.
5. Login to instructor account to see the result for the question.

**Expected behaviour**
Instructor should not see the response for that question as visibility is set to nobody. 

**Actual behaviour**
Instructor can see the response.

**Comment by damithc** [comment]
If this is indeed the case, that is a critical bug. Luckily, I don't think many instructors use this feature, but still, need to be fixed quickly. I'm surprised this is not covered in current tests 🤔

**Comment by joanneong** [comment]
Hmm...I tested on both Firefox and Chrome browsers using the `master` branch at commit `ada4896` but I am unable to reproduce this problem.

Since the steps above involved logging in to the student account to answer the survey, I tested only the various feedback paths that include "Students in this course will give feedback on...":

![image](https://user-images.githubusercontent.com/24493463/37094897-56e226ae-224f-11e8-9eae-2858f1087c24.png)

Following @Haozhe321 's steps, I also tried this setting:

![image](https://user-images.githubusercontent.com/24493463/37094921-6bb83654-224f-11e8-8b71-e7c7caabfb4b.png)

And then I logged out, logged in as a student, and answered the survey question:

![image](https://user-images.githubusercontent.com/24493463/37095008-b14f4f54-224f-11e8-8ef9-fa14048e666d.png)

![image](https://user-images.githubusercontent.com/24493463/37094999-a50c6394-224f-11e8-9fd2-9784f49e4df4.png)

But when I checked the results as an instructor:

![image](https://user-images.githubusercontent.com/24493463/37095032-c5f792b8-224f-11e8-8c6c-ea4c35e6852e.png)

So the behaviour is actually correct. For good measure, I also tried to edit student's responses via the instructor account (Moderate Responses) by having the student answer a different question in the same survey before this, and I could not see this private question nor the student's answer to it as well.

Looking at what has been merged since commit `49386fe`, I do not see any that might have affected this behaviour as well. @Haozhe321 Let me know if I missed out a step/misread something!

**Comment by joanneong** [comment]
Update: As discussed in person on last Thursday, the bug does not seem to be reproducible elsewhere. @bqnguyen94 can also verify this.

**Comment by bqnguyen94** [comment]
Yes, the problem seemed to present on @Haozhe321 's local database only.

**Comment by tran-tien-dat** [comment]
Maybe @Haozhe321 can upload your local database in `WEB-INF\appengine-generated\local_db.bin`?

**Comment by Haozhe321** [comment]
Hmm, the last we tried that with @joanneong it created some problems switching between databases. And also the database contains some of my email addresses so I think I shall not share it here. If the bug isn't reproducible then I guess it isn't that big a issue.

**Comment by damithc** [comment]
@Haozhe321 are you able to reproduce the problem for another instructor, starting from scratch?
